addon/components/frost-scroll.js

```diff
@@ -11,23 +10,37 @@ const {
 } = Ember
 import PropTypeMixin, {PropTypes} from 'ember-prop-types'
 
-const debouncePeriod = 150
-
 export default Component.extend(PropTypeMixin, {
 
-  // == Component properties ==================================================
+  // == Properties ============================================================
 
   classNames: ['frost-scroll'],
 
-  // == State properties ======================================================
-
   propTypes: {
     hook: PropTypes.string
   },
 
   // == Events ================================================================
 
-  initializeScroll: on('didInsertElement', function () {
+  didInsertElement () {
+    this._super(...arguments)
+    this._setupPerfectScroll()
+  },
+
+  willClearRender () {
```

Comment:
This is called before every re-render, I think you want http://emberjs.com/api/classes/Ember.Component.html#event_willDestroyElement